### ISSUE TITLE
macOS: Quick AutoFill is broken

### ISSUE THREAD

**Issue by TheFrenchGhosty** [keepassium/KeePassium#392]
**Description**

When using Quick AutoFill in Safari on macOS, the dialog will pop up, but when the KeePassium "box" open, nothing will work and it will be stuck on an empty screen, this box isn't possible to be closed, so Safari has to be quit.

Using "normal" AutoFill works, it only happens with Quick AutoFill.

**How to reproduce**

- Go to a website you have credentials for
- Try to use QuickAutofill

**Expected behavior**

It works (and the window can be closed)

**Screenshots**

![Screenshot-2024-10-12-17-13-52@2x](https://github.com/user-attachments/assets/975e020d-99a4-4f76-823c-b98e7bfd675c)

**Environment:**
 - Device: mac
 - OS: macOS Sonoma
 - App Version: 1.53 (latest)

**Additional context**

Maybe related to https://github.com/keepassium/KeePassium/issues/259

**Comment by feld** [comment]
Did you import a Keepass database that was created elsewhere? I had this happen and the issue was that the autofill functionality here has some limitations (memory usage??) and you need to change your Keepass database to the default Keepassium encryption settings: ChaCha20, Argon2d, 100 rounds, 1MB memory usage, 2 threads

That solved the problem for me

**Comment by TheFrenchGhosty** [comment]
@feld 

> Did you import a Keepass database that was created elsewhere?

I am using the same database as I'm using in KeePassXC yes, it was created in KeePassXC

>you need to change your Keepass database to the default Keepassium encryption settings: ChaCha20, Argon2d, 100 rounds, 1MB memory usage, 2 threads

I'd rather use the default from KeePassXC... those should work if the DB is unlocked already (it works on iPadOS)

**Comment by svezg** [comment]
Same issue here; it does not happen every time but more than half of the attempts.

**Comment by tibideky** [comment]
Same issue here, for me it happens every time when using Quickfill (autofill works ok without Quickfill)

**Comment by keepassium** [comment]
Hello everyone, sorry for the long silence. 

I can reproduce the issue. Apparently, AutoFill lifecycle has changed again, so the system now calls AutoFill methods in a different order than KeePassium expects. This had happened before and AutoFill is already a patchwork of conditional code that "mostly works" on iOS 15-18 and macOS 12-15. Fixing it with yet another patch would likely break things on older systems.

So our plan is to drop older systems, increasing the the minimum requirement to iOS 17 and macOS 14 (the ones we can actually test), then re-analyze the current AutoFill lifecycle, and rewire things accordingly. This is at the top of the queue, but might take some time to finish. Stay tuned!

**Comment by keepassium** [comment]
Just uploaded an update ([1.54.156](https://github.com/keepassium/KeePassium/releases/tag/1.54.156)), please give it a try.

**Comment by svezg** [comment]
> Just uploaded an update ([1.54.156](https://github.com/keepassium/KeePassium/releases/tag/1.54.156)), please give it a try.

Did some checking the past two days and it looks good; no hangups so far.

**Comment by tibideky** [comment]
Yes, I downloaded 1.54.156 via Testtflight, installed and seems to work well here again.
The only one-time issue I encountered was that the application crashed only once right after installing and setting up / adding the database and turning on quick-autofill, at the first try of using quick-autofill.
Since then, no crashes and quick-autofill works well and stable.

**Comment by matheusgomesms** [comment]
It seems it is fixed completely (1.54.156, latest MacOS), but I still managed to get this bug here and then. Not always, but like 10% of the cases. I'll check better what I'm doing differently, so others can reproduce too.

**Comment by keepassium** [comment]
Ok, I guess we can consider this fixed. Should this happen again, please comment to reopen. (Preferably with steps to reproduce :)
